server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -127,6 +137,12 @@ public ClientQuerySegmentWalker(
     this.serverConfig = serverConfig;
     this.cache = cache;
     this.cacheConfig = cacheConfig;
+    this.subqueryGuardrailHelper = new SubqueryGuardrailHelper(
+        lookupManager,
+        Runtime.getRuntime().maxMemory(),
+        httpClientConfig.getNumConnections()
```

Comment:
This is the # of connections from the Broker to Historicals. I think we should be using the # of queries, though? That's `ServerConfig#getNumThreads`.